homeassistant/components/thread/__init__.py

```diff
@@ -0,0 +1,220 @@
+"""The Thread integration."""
+from __future__ import annotations
+
+from http import HTTPStatus
+from json import JSONDecodeError
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.components.hassio import AddonError, AddonInfo, AddonManager
+from homeassistant.components.homeassistant_hardware.silabs_multiprotocol_addon import (
+    get_addon_manager,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import ConfigType
+
+from .models import OperationalDataSet, ThreadState
+
+DOMAIN = "thread"
+
+DATA_ADDON_MANAGER = "silabs_multiprotocol_addon_manager"
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Thread."""
+
+    return True
+
+
+REST_API_PORT = 8081
+
+
+class ThreadNetworkActiveError(HomeAssistantError):
+    """Raised on attempts to modify the active dataset when thread network is active."""
+
+
+class NoDatasetError(HomeAssistantError):
+    """Raised on attempts to update a dataset which does not exist."""
+
+
+async def _async_get_thread_rest_service_url(hass) -> str:
+    """Return Thread REST API URL."""
+    addon_manager: AddonManager = get_addon_manager(hass)
+    try:
+        addon_info: AddonInfo = await addon_manager.async_get_addon_info()
+    except (KeyError, AddonError) as err:
+        raise HomeAssistantError from err
+
+    if not addon_info.hostname:
+        raise HomeAssistantError
+
+    return f"http://{addon_info.hostname}:{REST_API_PORT}"
```

Comment:
Given that we can have multiple Thread add-ons, and even it could run in a remote container, I want to suggest that in a future we switch to relying on hassio discovery to send in the add-on URL for Thread discovery.